tasks/ast_tools/src/generators/typescript.rs

```diff
@@ -27,20 +27,30 @@ impl Generator for TypescriptGenerator {
         let estree_derive_id = codegen.get_derive_id_by_name("ESTree");
 
         let mut code = String::new();
+        let mut ast_node_names: Vec<String> = vec![];
         for type_def in &schema.types {
             if type_def.generates_derive(estree_derive_id) {
-                generate_ts_type_def(type_def, &mut code, schema);
+                generate_ts_type_def(type_def, &mut code, schema, &mut ast_node_names);
             }
         }
 
+        // manually append `FormalParameterRest`, which is generated via `add_ts_def`.
+        let ast_node_union = ast_node_names.join(" | ");
+        write_it!(code, "export type Node = {ast_node_union} | FormalParameterRest;\n\n");
+
         Output::Javascript { path: TYPESCRIPT_DEFINITIONS_PATH.to_string(), code }
     }
 }
 
 /// Generate Typescript type definition for a struct or enum.
 ///
 /// Push type defs to `code`.
-fn generate_ts_type_def(type_def: &TypeDef, code: &mut String, schema: &Schema) {
+fn generate_ts_type_def(
+    type_def: &TypeDef,
+    code: &mut String,
+    schema: &Schema,
+    ast_node_names: &mut Vec<String>,
+) {
```

Comment:
Nit: Could you put `ast_node_names` param before `schema`. Everywhere throughout `oxc_ast_tools` we always have `schema` as last param. I think it's preferable to maintain that consistency.